src/Sentry/Internal/Hub.cs

```diff
@@ -156,6 +163,45 @@ public void BindException(Exception exception, ISpan span)
 
         public SentryTraceHeader? GetTraceHeader() => GetSpan()?.GetTraceHeader();
 
+        public void StartSession()
+        {
+            try
+            {
+                var session = _sessionManager.StartSession();
+                if (session is not null)
+                {
+                    ConfigureScope(scope => scope.Session = session);
+                    CaptureSession(session.CreateUpdate(true));
+                }
+            }
+            catch (Exception ex)
+            {
+                _options.DiagnosticLogger?.LogError(
+                    "Failed to start a session.",
+                    ex
+                );
+            }
+        }
+
+        public void EndSession(SessionEndStatus status = SessionEndStatus.Exited)
+        {
+            try
+            {
+                var session = _sessionManager.EndSession(status);
```

Comment:
We need to sync between this and startSession and `ConfigureScope(s => s.Session`